build/hellobuild.md

```diff
@@ -0,0 +1,109 @@
+* **CMD \<command\>** - letting you define the default program that is run once you start the container based on this image. Each Dockerfile only has one CMD, and only the last CMD instance is respected when multiple exist.
+Dockerfiles are crucial inputs for image builds and can facilitate automated, multi-layer image builds based on your unique configurations. Dockerfiles can start simple and grow with your needs and support images that require complex instructions.
+For all the possible instructions, see the [Dockerfile reference](https://docs.docker.com/engine/reference/builder/).
+
+## Example
+Here’s a simple Dockerfile example to get you started in building an image. We’ll take a simple Hello World python flask application, and bundle it into a docker image that we can test locally or deploy anywhere!
+
+Let’s say we have the following in a `hello.py` file in our local directory (sample A):
+
+{% highlight python lineos %}
+from flask import Flask
+app = Flask(__name__)
+
+@app.route("/")
+def hello():
+    return "Hello World!"
+{% endhighlight %}
+
+Don’t worry about understanding the full example if you’re not familiar with Python - it’s just a simple web server that will contain a single page that says “Hello World”. If you’re copying or writing out the example yourself, make sure to copy over the indentation as well!
+
+Here’s a Dockerfile that Docker Build will use to create an image for our application (sample B): 
+
+{% highlight dockerfile lineos %}
+# syntax=docker/dockerfile:1
+FROM ubuntu:22.04
+
+# install app dependencies
+RUN apt-get update && apt-get install -y python3 python3-pip
+RUN pip install flask
+
+# install app
+COPY hello.py /
+
+# final configuration
+ENV FLASK_APP=hello
+ 	EXPOSE 8000
+CMD flask run --host 0.0.0.0 --port 8000
+{% endhighlight %}
+
+* Line 1 - This is our syntax directive. It pins the exact version of the dockerfile syntax we’re using. As a [best practice](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/), this should be the very first line in all our Dockerfiles as it informs Buildkit the right version of the Dockerfile to use. See also [Syntax](https://docs.docker.com/engine/reference/builder/#syntax).
+    > **Note:** Initiated by a # like regular comments, this line is treated as a >directive when you are using Buildkit (default), otherwise it is ignored.
+
+
+* Line 2 - Here FROM sets our base image to the 22.04 release of Ubuntu. All following instructions are executed on this base image, in this case, a Ubuntu environment. 
+The notation ubuntu:22:04, follows the name:tag standard for naming Docker images. 
+When you build your image you use this notation to name your images and use it to specify any existing docker image.
+There are many public images you leverage in your projects. Explore [Docker Hub](https://hub.docker.com/_/ubuntu/?tab=tags) to find out.
+
+* Line 4 - Comments in dockerfiles begin with the # symbol. 
```

Comment:
```suggestion
* **# install app dependencies** - Comments in dockerfiles begin with the # symbol. 
```

Reply:
Maybe for now but less confusing that line numbers to me

Reply:
Agreed, changing for all lines in the sample.